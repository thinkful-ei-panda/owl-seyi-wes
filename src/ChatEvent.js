import React from 'react';

function ChatEvent(props){

    const person=props.participants.find(participant => participant.id===props.chatEvent.participantId)
    const date=new Date(props.chatEvent.timestamp);
    let adjustedTime=date.toLocaleTimeString().split('');
    adjustedTime.splice(4,3);
    adjustedTime=adjustedTime.join('');
    switch (props.chatEvent.type){
        case 'join':
            return  <div className='chat-event-container2 center'>
                        <div>{adjustedTime}</div>
                        <div className='event'>{person.name} joined</div>
                    </div>
        case 'leave':
            return  <div className='chat-event-container2 center'>
                        <div>{adjustedTime}</div>
                        <div className='event'>{person.name} left the session</div>
                    </div>
        case 'thumbs-up':
            return  <div className='chat-event-container2 center'>
                        <div>{adjustedTime}</div>
                        <div className='event'>{person.name} gave a thumbs up</div>
                    </div>
        case 'thumbs-down':
            return  <div className='chat-event-container2 center'>
                        <div>{adjustedTime}</div>
                        <div className='event'>{person.name} gave a thumbs down</div>
                    </div>
        case 'raise-hand':
            return  <div className='chat-event-container2 center'>
                        <div>{adjustedTime}</div>
                        <div className='event'>{person.name} raised their hand</div>
                    </div>
        case 'clap':
            return  <div className='chat-event-container2 center'>
                        <div>{adjustedTime}</div>
                        <div className='event'>{person.name} clapped</div>
                    </div>
        case 'join-stage':
            return  <div className='chat-event-container2 center'>
                        <div>{adjustedTime}</div>
                        <div className='event'>{person.name} joined the stage</div>
                    </div>
        case 'leave-stage':
            return  <div className='chat-event-container2 center'>
                        <div>{adjustedTime}</div>
                        <div className='event'>{person.name} left the stage</div>
                    </div>
        case 'message':
            return  <div className='chat-event-container'>
                        <div className='small'><img src={person.avatar} alt='avatar'/></div>
                        <div>
                            <div>{person.name}<span className='time'>{adjustedTime}</span></div>
                            <div>{props.chatEvent.message}</div>
                        </div>
                    </div>
        default:
            return <></>
    }
}

export default ChatEvent