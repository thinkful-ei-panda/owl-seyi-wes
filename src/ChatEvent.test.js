import React from 'react';
import ReactDOM from 'react-dom';
import ChatEvent from './ChatEvent';
import store from './store'

import renderer from 'react-test-renderer';

//<ChatEvent key={index} participants={props.participants} chatEvent={props.chatEvents[index]} />
describe('ChatEvent', () => {

    const props={
        key:0,
        participants:store.participants,
        chatEvent:{
            participantId: 2,
            type: 'thumbs-up',
            timestamp: 1548852484247
        },
    }
  it('renders without crashing', () => {

    const div = document.createElement('div');
    ReactDOM.render(<ChatEvent {...props}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ChatEvent {...props}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

});

