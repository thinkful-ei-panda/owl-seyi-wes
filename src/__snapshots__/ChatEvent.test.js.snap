// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatEvent renders the UI as expected 1`] = `
<div
  className="chat-event-container2 center"
>
  <div>
    4:48 AM
  </div>
  <div
    className="event"
  >
    Caty Flucker
     gave a thumbs up
  </div>
</div>
`;
